### ISSUE TITLE
Improve load time of registration page for competition admins

### ISSUE THREAD

**Issue by JonEsparaz** [thewca/worldcubeassociation.org#5113]
**Is your feature request related to a problem? Please describe.**
The registration page for competition admins has a (somewhat) annoyingly long load time. Example: WCA North American Championship 2020 takes ~15 seconds for ~450 total competitors.

**Describe the solution you'd like**
Don't load Approved registrations and Deleted registrations unless requested by the user. 

**Describe alternatives you've considered**
Perhaps has the fixes in #731 could provide an alternate solution. (I admittedly have no experience with Rails, so I'm unsure if this fix is applicable).

**Comment by viroulep** [comment]
450 registrations is "nothing", it definitely means something is wrong on our side before having to consider to not show some registrations.

Simply looking at the log of a load of the registrations page of a competition with active payments shows a silly hidden n+1 query that I fixed in #5113.
It's unfortunate but not that surprising that we did not notice it before, as the developer dump we use locally has no competitions with payments enable (because we strip information about stripe accounts).
